File: src/hsrb_samples/hsrb_mounted_devices_samples/src/hsrb_mounted_devices_samples/status_led/change_led_color.cpp
/// @brief Change Status LED Color Sample
/// @brief Copyright (C) 2017 Toyota Motor Corporation
#include <string>

#include <ros/ros.h>
#include <std_msgs/ColorRGBA.h>

namespace {
const uint16_t kLEDInputDataSize = 256;
const float kDefaultColorG = 0.2;
const float kDefaultColorB = 0.6;
}  // unnamed namespace

int main(int argc, char** argv) {
  ros::init(argc, argv, "hsrb_change_led_color");

  // Create Publisher to change status led color
  ros::NodeHandle nh;
  std::string status_led_topic = "/hsrb/command_status_led_rgb";
  ros::Publisher led_pub =
      nh.advertise<std_msgs::ColorRGBA>(status_led_topic, 100);

  // Wait for connection
  while (led_pub.getNumSubscribers() == 0) {
    ros::Duration(0.1).sleep();
  }

  // 50Hz is enough frequency to do gradual color change
  ros::Rate rate(50);

  // Value g and b are fixed
  std_msgs::ColorRGBA color;
  color.g = kDefaultColorG;
  color.b = kDefaultColorB;

  // Value r changes gradually
  for (uint16_t num = 0; num < kLEDInputDataSize; num++) {
    color.r = num / static_cast<float>(kLEDInputDataSize - 1);
    led_pub.publish(color);
    rate.sleep();
  }
  return EXIT_SUCCESS;
}

File: src/hsrb_samples/hsrb_mounted_devices_samples/src/hsrb_mounted_devices_samples/status_led/change_led_color.py
#!/usr/bin/env python
# Copyright (C) 2017 Toyota Motor Corporation
"""Change Status LED Color Sample"""
import rospy
from std_msgs.msg import ColorRGBA

_LED_INPUT_DATA_SIZE = 256
_DEFAULT_COLOR = ColorRGBA(g=0.2, b=0.6)


def main():
    # Create Publisher to change status led color
    status_led_topic = '/hsrb/command_status_led_rgb'
    led_pub = rospy.Publisher(status_led_topic,
                              ColorRGBA, queue_size=100)

    # Wait for connection
    while led_pub.get_num_connections() == 0:
        rospy.sleep(0.1)

    # 50Hz is enough frequency to do gradual color change
    rate = rospy.Rate(50)

    # Value r changes gradually (value g and b are fixed)
    color = _DEFAULT_COLOR
    for num in range(_LED_INPUT_DATA_SIZE):
        color.r = num / float(_LED_INPUT_DATA_SIZE - 1)
        led_pub.publish(color)
        rate.sleep()


if __name__ == '__main__':
    rospy.init_node('hsrb_change_led_color')
    main()
